packages/cli/src/__tests__/commands/Init.test.ts

```diff
@@ -58,6 +58,26 @@ test('works with provider param - postgresql', async () => {
   `)
 })
 
+test('works with provider param - cockroachdb', async () => {
+  ctx.fixture('init')
+  const result = await ctx.cli('init', '--datasource-provider', 'cockroachdb')
+  expect(stripAnsi(result.stdout)).toMatchSnapshot()
+
+  const schema = fs.readFileSync(join(ctx.tmpDir, 'prisma', 'schema.prisma'), 'utf-8')
+  expect(schema).toMatch(defaultSchema('cockroachdb'))
```

Comment:
The default schema needs an extra line for the generator like 
```
previewFeatures = ["cockroachdb"]
```

I can suggest doing it like we did for mongodb and removed here
https://github.com/prisma/prisma/pull/12660/files#diff-3a2fd0c2f24931daa62a694bc42215135099cded6cc9da76ef7c64db4fb6f10fL21-L39